### PR TITLE
Revise KIC plugin and auth guides for GWAPI

### PR DESCRIPTION
**Pull request by rainest**
### Summary
Continuing the addition of Gateway API content and reusable includes for common tasks from https://github.com/Kong/docs.konghq.com/pull/4766 onward, into the plugin and authentication guides, and then some more popular guides.

Add a blurb about KongIngress support for HTTPRoute.

Add support for providing a service name to the routing configuration include.

Updates all the KIC guide Kong versions in responses to 3.1.1, since edits over time have made these get out of sync within guides.

### Reason
Part of https://github.com/Kong/kubernetes-ingress-controller/issues/2878 and https://github.com/Kong/kubernetes-ingress-controller/issues/2966

### Testing

https://deploy-preview-4894--kongdocs.netlify.app/kubernetes-ingress-controller/latest/guides/configure-acl-plugin/
https://deploy-preview-4894--kongdocs.netlify.app/kubernetes-ingress-controller/latest/guides/using-kongplugin-resource/
https://deploy-preview-4894--kongdocs.netlify.app/kubernetes-ingress-controller/latest/guides/using-consumer-credential-resource/
https://deploy-preview-4894--kongdocs.netlify.app/kubernetes-ingress-controller/latest/guides/using-external-service/
https://deploy-preview-4894--kongdocs.netlify.app/kubernetes-ingress-controller/latest/guides/using-kongingress-resource/#set-up-ingress

Accidentally deleted ACL's log, but it is tested and fixed. Only went through GWAPI for these, since the Ingress content from #4766 is largely unchanged (I did a spot check to confirm that it's not broken by the addition of the `service` parameter).

[ext.txt](https://github.com/Kong/docs.konghq.com/files/10241409/ext.txt)
[plugin.txt](https://github.com/Kong/docs.konghq.com/files/10241410/plugin.txt)
[consumer.txt](https://github.com/Kong/docs.konghq.com/files/10241411/consumer.txt)

### CODE REVIEW

**Comment by Guaris** [comment]
When this is ready please feel free to remove the draft status so we can approve it.

**Comment by rainest** [comment]
@Guaris this depends on https://github.com/Kong/docs.konghq.com/pull/4766; it needs to be merged, and this needs to be rebased, before this can proceed.

If it works better for ya'll I could make this the only PR, containing the commits and changes from both. I'd kept them separate since they're both fairly large on their own already.

**Comment by rainest** [comment]
Still needs an E2E command validation pass from me, but going ahead and marking ready for review of the copy and templating.

**Comment by rainest** [comment]
Vale is unhappy for reasons I cannot entirely determine. It's apparently detecting terminology issues on a bunch of content I _didn't_ edit. Didn't find any `app/_src/ku` in the log, so it looks like the content here is fine :shrug:

**Review by cloudjumpercat — CHANGES_REQUESTED** [review]
Left a few suggestions. I think Vale is failing because you made an update in the file, so Vale checks each file that is touched. I think if you just fix the "Preserving" text, it should pass.

File: app/_src/kubernetes-ingress-controller/guides/cert-manager.md [hunk 1]
```diff
@@ -154,7 +154,7 @@ Date: Fri, 21 Jun 2019 21:14:45 GMT
 Server: echoserver
 X-Kong-Upstream-Latency: 1
 X-Kong-Proxy-Latency: 1
-Via: kong/1.1.2
+Via: kong/3.1.1
```
[the remark this answers is not in the document]
Comment (rainest) [reply app/_src/kubernetes-ingress-controller/guides/cert-manager.md:157]: There are tags for it, so AFAICT it is: https://hub.docker.com/_/kong/tags?page=1&name=3.1.1

**Pull request merged**